mithril-aggregator/src/services/epoch_service.rs

```diff
@@ -671,82 +722,87 @@ mod tests {
-    async fn build_service(
-        epoch: Epoch,
-        current_epoch_fixture: &MithrilFixture,
-        additional_params: &[ServiceBuilderParameters],
-    ) -> MithrilEpochService {
-        let signer_retrieval_epoch = epoch.offset_to_signer_retrieval_epoch().unwrap();
-        let next_signer_retrieval_epoch = epoch.offset_to_next_signer_retrieval_epoch();
-        let mut next_epoch_fixture = current_epoch_fixture;
-        let mut upcoming_protocol_parameters = current_epoch_fixture.protocol_parameters();
-        let mut future_protocol_parameters = current_epoch_fixture.protocol_parameters();
-
-        for params in additional_params {
-            match params {
-                ServiceBuilderParameters::DifferentFixtureForSecondEpoch(fixture) => {
-                    next_epoch_fixture = fixture
-                }
-                ServiceBuilderParameters::UpcomingProtocolParameters(params) => {
-                    upcoming_protocol_parameters = params.clone()
-                }
-                WithFutureProtocolParameters(params) => future_protocol_parameters = params.clone(),
-            }
-        }
-
-        let epoch_settings_storer = FakeEpochSettingsStorer::new(vec![
-            (
-                signer_retrieval_epoch,
-                AggregatorEpochSettings {
+    impl EpochServiceBuilder {
+        fn new(epoch: Epoch, current_epoch_fixture: MithrilFixture) -> Self {
+            let next_epoch_fixture = current_epoch_fixture.clone();
+            Self {
+                // Aggregator configuration
+                cardano_transactions_signing_config: CardanoTransactionsSigningConfig::dummy(),
+                future_protocol_parameters: current_epoch_fixture.protocol_parameters(),
+                network: CardanoNetwork::TestNet(0),
+                allowed_discriminants: BTreeSet::new(),
+                // Epoch used for verification keys and database
+                signer_retrieval_epoch: epoch.offset_to_signer_retrieval_epoch().unwrap(),
+                next_signer_retrieval_epoch: epoch.offset_to_next_signer_retrieval_epoch(),
+                // Signers in verification key store
+                signers_with_stake: current_epoch_fixture.signers_with_stake(),
+                next_signers_with_stake: next_epoch_fixture.signers_with_stake(),
+                // Database data
+                stored_epoch_settings: AggregatorEpochSettings {
                     protocol_parameters: current_epoch_fixture.protocol_parameters(),
                     cardano_transactions_signing_config: CardanoTransactionsSigningConfig::dummy(),
                 },
-            ),
-            (
-                next_signer_retrieval_epoch,
-                AggregatorEpochSettings {
-                    protocol_parameters: next_epoch_fixture.protocol_parameters(),
+                stored_next_epoch_settings: AggregatorEpochSettings {
+                    protocol_parameters: current_epoch_fixture.protocol_parameters(),
```

Comment:
It looks like it is not computed on the correct fixture: 
```suggestion
                    protocol_parameters: next_epoch_fixture.protocol_parameters(),
```

Reply:
Done ✅